client/src/app/shared/lab-card/lab-card.component.html

```diff
@@ -4,7 +4,9 @@
       <h3>{{ lab.name | words: 5 }}</h3>
       <span class="ml-user-profile-lab-card-private-badge" *ngIf="lab.is_private">Private</span>
       <div class="ml-lab-card-user" *ngIf="user && showUsername">
-        by <a class="ml-lab-card-username" title="{{ user.displayName }}'s profile" [routerLink]="['/user', user.id]">{{user.displayName}}</a>
+        by
+        <img *ngIf="user.photoUrl && showUserAvatar" [src]="user.photoUrl" alt="User avatar">
+        <a class="ml-lab-card-username" title="{{ user.displayName }}'s profile" [routerLink]="['/user', user.id]">{{user.displayName}}</a>
```

Comment:
Two things here:

1. I realized we have a bug in this component (not introduced by you). We can't have `<a>` inside `<a>` tags. This not only breaks the web but also the user experience. Try it out yourself by clicking on the username, you'll always end up on the actual lab. This is because the event is triggered by the outer tag.
There's several ways to fix this
  - Change the inner link to a button and use [event delegation](https://davidwalsh.name/event-delegate). That means the outer link needs to become a button as well, so we can attach a handler on it that prevents the link from performing a location change. The downside is that we loose the semantics of links here, which I'd like to avoid.
  - We could change the lab card in the sense that not the whole lab card title is a link and therefore clickable, but rather introduce multiple links inside of the whole title area. For example, the lab name would be a link, the arrow icon would become a link and the user name would be a link. [There's some CSS tricks that enable that](https://stackoverflow.com/questions/36546546/valid-alternative-to-nested-links-a-tags-for-rich-web-applications). Obviously, the goal here would be to keep the whole title area clickable
  - We could also just  go back to having those three links without having the whole title area clickable. That would probably be the easiest solution, but introduce a user experience regression if you will.

/cc'ing @d3lm here for his opinion

2. I don't think we need `showUserAvatar` as a flag here. We need it for `showUsername` so we can configure whether we want to show a username or not. For example, in the user-profile page, we don't wanna show the user name as we know, we're in the profile of that user. I don't think we want that control for the avatar. In addition to that, if `showUsername` is falls, the avatar isn't inserted either because it's a child element of `.ml-lab-card-user`, so we kind of get implicit control.

Reply:
Hello @PascalPrecht, thanks for the detailed feedback 😀! 

1. I'll have a deeper look at what's going on there with the routing because it should work as expected (for example, if you remove the `[routerLink]`s on the `<a>`s and add 2 `href`s to some random external sites, say outer is google and inner is bing it should route you as expected). I've tried to make a simmilar [quick example](https://stackblitz.com/edit/ng2-nested-anchors) of this in stackblitz with nested routerLinks and it seems to work as expected. 

2. sure, i'll remove it, it just made sense for me at the time to have control over showing the avatar as well since you had other options as well but you're right that the avatar is somewhat linked with the username, so `showUsername` should be enough. 

I'll update the code and keep you posted on the routing problem.

Reply:
> I've tried to make a simmilar quick example of this in stackblitz with nested routerLinks and it seems to work as expected.

I'm surprised this works! Even though it does, I don't think we want to break semantics here. Nested links aren't good for SEO either. So I'd favour to choose one of the other options.

Reply:
Yep, agreed. I was just caught up in the technical details of why it wasn't working, but for sure we'll find a better way than nesting the `<a>`s.